### PR TITLE
[OpenAI-Assistant] Init PR to add feature branch to main branch.

### PR DESCRIPTION
**Pull request by mssfang**
This is the initial PR to feature branch to main branch for OpenAI Assistant.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -111,13 +111,13 @@
 /sdk/eventhubs/                                      @conniey @anuchandy @lmolkova
 
 # PRLabel: %Event Hubs
-/sdk/eventhubs/microsoft-azure-eventhubs-eph/        @JamesBirdsall @sjkwak
+/sdk/eventhubs/microsoft-azure-eventhubs-eph/        @sjkwak
```
Comment (mssfang) [inline .github/CODEOWNERS:114]: @conniey @anuchandy Seems James no longer in Micrsoft. Can I safely delete it? Somehow it is only code owner failed my PR pipeline even there are more invalid users.

Comment (JimSuplizio) [reply .github/CODEOWNERS:114]: @mssfang I fixed this in main, just pull the updated CODEOWNERS into your branch and you'll be good to go.

Comment (mssfang) [reply .github/CODEOWNERS:114]: Thanks Jim

File: sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java [hunk 2]
```diff
@@ -0,0 +1,3054 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import reactor.core.publisher.Mono;
+
+/**
+ * Initializes a new instance of the asynchronous AssistantsClient type.
+ */
+@ServiceClient(builder = AssistantsClientBuilder.class, isAsync = true)
+public final class AssistantsAsyncClient {
```
Comment (srnagar) [inline sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java:65]: The client name is too generic. We should consider naming this `OpenAIAssistantsAsyncClient`.

Comment (mssfang) [reply sdk/openai/azure-ai-openai-assistants/src/main/java/com/azure/ai/openai/assistants/AssistantsAsyncClient.java:65]: We can bring this naming question to archboard. All languages go with generated name for the first beta.
https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/openai/Azure.AI.OpenAI.Assistants#authenticate-the-client

**Review by srnagar — APPROVED** [review]

**Pull request merged**